rx-preferences/src/main/java/com/f2prateek/rx/preferences2/RealPreference.java

```diff
@@ -65,22 +67,21 @@
     return adapter.get(key, preferences);
   }
 
-  @Override public synchronized void set(@Nullable T value) {
+  @Override public synchronized void set(@NonNull T value) {
+    checkNotNull(value, "value == null");
     SharedPreferences.Editor editor = preferences.edit();
-    if (value == null) {
-      editor.remove(key);
-    } else {
-      adapter.set(key, value, editor);
-    }
+    adapter.set(key, value, editor);
     editor.apply();
   }
 
-  @Override public boolean isSet() {
+  @Override public synchronized boolean isSet() {
```

Comment:
Lets say thread A calls `isSet` and depending on the result writes writes different values. In the meantime Thread B writes a value and now we have a classic race condition.